manta-trusted-setup/src/groth16/mpc.rs

```diff
@@ -48,20 +56,62 @@ use {
     serde(crate = "manta_util::serde", deny_unknown_fields)
 )]
 #[derive(derivative::Derivative)]
-#[derivative(Clone(bound = ""))]
+#[derivative(Clone(bound = ""), Debug(bound = ""))]
 pub struct State<P>(
     #[cfg_attr(
         feature = "serde",
         serde(
-            serialize_with = "canonical_serialize::<ProvingKey<P::Pairing>, _>",
-            deserialize_with = "canonical_deserialize::<'de, _, ProvingKey<P::Pairing>>"
+            serialize_with = "canonical_serialize_uncompressed::<ProvingKey<P::Pairing>, _>",
+            deserialize_with = "canonical_deserialize_unchecked::<'de, _, ProvingKey<P::Pairing>>"
```

Comment:
Why use `...uncompressed` for serialize but `...unchecked` for deserialize?

Reply:
The reason to use `deserialize_unchecked` is to prevent the client from performing on-curve/in-subgroup checks for the data it receives from the server -- this slows down the client too much, and we can consider the server to be trustworthy. We will add a manual check on the server side, since data from the client ought to be checked.  
  
Arkworks `deserialize_unchecked` assumes the serialization is uncompressed, which is why we need to use `serialize_uncompressed`